### PR TITLE
Explore page port

### LINKED ISSUE #273: Port `/anime/explore` page
Port from:
https://github.com/coreproject-moe/CoreProject/blob/4f23d3031843cafe7cdd613acfcd5f456a36ccb6/backend/django_core/static_src/components/page/explore/Index.svelte

### PR DESCRIPTION
**Pull request by meowsocute**
`/anime/explore/` page ported to Svelte 5,
> added `loadsh-es` dependency.
> added `/types/anime.ts`, `/functions/get_csrf_token.ts`, `/functions/get_csrf_token.ts`.

Fix #273

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
 		"@tailwindcss/typography": "^0.5.13",
 		"@types/chroma-js": "^2.4.4",
 		"@types/eslint": "^9.6.0",
+		"@types/lodash-es": "^4.17.12",
```
Comment (baseplate-admin) [inline package.json:37]: We dont want lodash

File: package.json [hunk 2]
```diff
@@ -108,6 +109,8 @@
 	"dependencies": {
 		"@vercel/analytics": "^1.3.1",
 		"@vercel/speed-insights": "^1.0.12",
+		"htmx.org": "^1.9.12",
+		"lodash-es": "^4.17.21",
```
Comment (baseplate-admin) [inline package.json:113]: these 2 are not needed?

File: src/lib/components/anime_card.svelte [hunk 3]
```diff
@@ -0,0 +1,145 @@
+<script lang="ts">
+	import { enhance_anchor } from "$functions/anchor_enhancements";
```
Comment (baseplate-admin) [inline src/lib/components/anime_card.svelte:2]: This function wont work in svelte environment

File: src/lib/components/anime_card.svelte [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	import ScrollArea from "./scroll_area.svelte";
+
+    export let anime_mal_id: number;
+    export let anime_name: string;
+    export let anime_image: string;
+    export let anime_total_episodes: number | null;
+    export let anime_synopsis: string;
+    export let anime_studios: Array<{ name: string }>;
+    export let anime_genres: Array<{ name: string }>;
+    export let anime_rating: string;
```
Comment (baseplate-admin) [inline src/lib/components/anime_card.svelte:17]: Neither of them are runes

**Review by baseplate-admin — CHANGES_REQUESTED** [review]
Overall a solid start

File: src/lib/functions/anchor_enhancements.ts
Comment (moonlitgrace) [inline src/lib/functions/anchor_enhancements.ts:1]: this function is not needed for svelte

File: src/lib/functions/get_csrf_token.ts
Comment (moonlitgrace) [inline src/lib/functions/get_csrf_token.ts:1]: this too

File: src/lib/functions/urls/goto.ts
Comment (moonlitgrace) [inline src/lib/functions/urls/goto.ts:1]: this too

File: src/lib/functions/urls/reverse.ts
Comment (moonlitgrace) [inline src/lib/functions/urls/reverse.ts:1]: and this too

**Review by moonlitgrace — COMMENTED** [review]
:standing_man:  :gun:

File: src/routes/anime/+layout.svelte [hunk 5]
```diff
@@ -335,8 +336,9 @@
 					clip-rule="evenodd"
 					d="M24 0H0V24C0 10.7451 10.7452 0 24 0Z"
 				/>
-			</svg>
-			<div id="page" class="w-full overflow-y-scroll">
```
Comment (moonlitgrace) [inline src/routes/anime/+layout.svelte:339]: dont remove it here, btw it was working fine on firefox
maybe you try it on ff?

https://coreproject-v4-ebfwbikcy-baseplateadmins-projects.vercel.app/anime/mal/1/episode/1

**Review by moonlitgrace — CHANGES_REQUESTED** [review]

File: src/routes/anime/explore/+page.svelte [hunk 6]
```diff
@@ -143,7 +143,8 @@
 		};
 		for (const [key, val] of Object.entries(search_map)) {
 			if (!isEmpty(val) && !isUndefined(val)) {
-				url.searchParams.set(key, val);
+				const value = typeof val === "string" ? val : "";
+				url.searchParams.set(key, value);
```
Comment (moonlitgrace) [inline src/routes/anime/explore/+page.svelte:147]:
```ts
url.searchParams.set(key, String(va));
```

and can you tell why you need to ensure if this a string? it is already not?

**Comment by moonlitgrace** [comment]
Porting done on #477

**Pull request closed without merging**